packages/mds-ingest-migration-processor/processor.ts

```diff
@@ -30,49 +32,147 @@ import {
+          logger.info(`Migrated device ${migrated.device_id}`)
+        }
+      } catch (error) {
+        logger.error(`Error migrating device`, { topic, message })
+      }
+    },
+    shutdown: async () => undefined
+  }
+}
+
+const DevicesMigrationProcessor = StreamProcessor<Device_v0_4_1 & { id: number }, MigratedDevice>(
   MigrationDataSource('device'),
-  async device => convert_v1_0_0_device_to_1_1_0(convert_v0_4_1_device_to_1_0_0(device)),
-  [],
+  async ({ id, ...device }) => ({
+    device: convert_v1_0_0_device_to_1_1_0(convert_v0_4_1_device_to_1_0_0(device)),
+    migrated_from: MigratedFrom('device', id)
+  }),
+  [IngestServiceMigratedDeviceSink()],
   [MigrationErrorSink('device')]
 )
 
-const EventsMigrationProcessor = StreamProcessor<VehicleEvent_v0_4_1 & { id: number }, VehicleEvent_v1_1_0>(
+interface MigratedVehicleEvent {
+  event: VehicleEvent_v1_1_0
+  migrated_from: MigratedEntityModel
+}
+
+const IngestServiceMigratedVehicleEventSink = (): StreamSink<MigratedVehicleEvent> => () => {
+  const topic = MigrationSourceTopic('event')
+  return {
+    initialize: async () => undefined,
+    write: async message => {
+      try {
+        const [{ event, migrated_from }] = asArray(message)
+        const migrated = await IngestServiceClient.writeMigratedVehicleEvent(event, migrated_from)
+        if (migrated) {
+          logger.info(`Migrated event ${migrated.device_id}/${migrated.timestamp}`)
+        }
+      } catch (error) {
+        logger.error(`Error migrating event`, { topic, message })
+      }
+    },
+    shutdown: async () => undefined
+  }
+}
+
+const EventsMigrationProcessor = StreamProcessor<
+  VehicleEvent_v0_4_1 & { id: number; service_area_id: Nullable<UUID> },
+  MigratedVehicleEvent
+>(
   MigrationDataSource('event'),
-  async event => convert_v1_0_0_vehicle_event_to_v1_1_0(convert_v0_4_1_vehicle_event_to_v1_0_0(event)),
-  [],
+  async ({ id, service_area_id, ...event }) => ({
+    event: convert_v1_0_0_vehicle_event_to_v1_1_0(convert_v0_4_1_vehicle_event_to_v1_0_0(event)),
+    migrated_from: MigratedFrom('event', id)
+  }),
+  [IngestServiceMigratedVehicleEventSink()],
   [MigrationErrorSink('event')]
 )
 
-const TelemetryMigrationProcessor = StreamForwarder<Telemetry & { id: number }>(
+interface MigratedTelemetry {
+  telemetry: Telemetry
+  migrated_from: MigratedEntityModel
+}
+
+const IngestServiceMigratedTelemetrySink = (): StreamSink<MigratedTelemetry> => () => {
+  const topic = MigrationSourceTopic('telemetry')
+  return {
+    initialize: async () => undefined,
+    write: async message => {
+      try {
+        const [{ telemetry, migrated_from }] = asArray(message)
+        const migrated = await IngestServiceClient.writeMigratedTelemetry(telemetry, migrated_from)
+        if (migrated) {
+          logger.info(`Migrated telemetry ${migrated.device_id}/${migrated.timestamp}`)
+        }
+      } catch (error) {
+        logger.error(`Error migrating telemetry`, { topic, message })
```

Comment:
I believe unless you `throw` here, this won't get handled by the error handling sink